include/freerdp/codec/h264.h

```diff
@@ -60,6 +70,12 @@ extern "C"
 	                                  const RECTANGLE_16* regionRect, BYTE** ppDstData,
 	                                  UINT32* pDstSize, RDPGFX_H264_METABLOCK* meta);
 
+	/* API for user to fill YUV I420 buffer before encoding */
+	FREERDP_API INT32 avc420_get_yuv_buffer(H264_CONTEXT* h264, UINT32 nSrcStride,
+					UINT32 nSrcWidth, UINT32 nSrcHeight,
+					BYTE* YUVData[3], int stride[3]);
```

Comment:
`stride` should be `size_t` (or another unsigned type)

Reply:
also, not sure this is a good way to utilize the encoder.
wouldn't it be better to keep the `YUV` buffers in the callers scope and provide them directly with the encode function?

Reply:
I also thought about this alternative approach. YUV buffers are currently managed by `H264_CONTEXT`. If we were to take over buffer management, the logic of `avc420_ensure_buffer` needs to be replicated, partially with the help of `swscale` library, but it is still extra work.

Reply:
ok, we can still refactor later on as long as we don´t tag this :grin: